all/test.sh

```diff
@@ -24,15 +26,21 @@ testGeneration() {
     extra_args=$@
     echo "Testing language $lang $expected_output_dir $extra_args"
 
+    mkdir -p "$name"
+    cp -r ./all "./$name"
+    pushd "./$name"
+
     # Test calling a file directly.
     exitCode=0
-    docker run --rm -v=`pwd`:/defs $CONTAINER -f all/test/test.proto -l $lang -i all/test/ $extra_args > /dev/null || exitCode=$?
+    docker run --rm -v=$(pwd):/defs "$CONTAINER" -f all/test/test.proto -l "$lang" -i all/test/ $extra_args > /dev/null || exitCode=$?
```

Comment:
Codacy is yelling about this line, but `$extra_args` can't be quoted - it needs to stay as an array, or the arguments don't get passed to docker correctly

Reply:
`extra_args` doesn't look like it's defined as an array to me. Line 26 should probably be `extra_args=( "$@" )` to make it an array.

Then, instead of `$extra_args`, Codacy might be okay with `"${extra_args[@]}`.

Reply:
Yeah, if you don't need it to be an array, leave it as-is, and just tell Codacy to ignore that issue, is my suggestion, based on what you're saying.

Instead of an array, I'd call it something like "a space-separated string of arguments", FWIW.

Reply:
Fair - once I saw your feedback, I read up on the `$@` and that is indeed what it is - so TIL it isn't an array like in normal programming languages 😄 
If I were an admin in this repo, I'd definitely do that. Unfortunately, I don't have the rights to make codacy ignore it.

Reply:
I don't have the rights to do that either, but somebody here must!

That being said, it has always struck me as a code smell that wildcard-matching on a variable containing args (which are subject to unwanted globbing, besides the quoting issue that is at least constrained by the callers) is used to "parse" those arguments.

It'd be great to refactor that code to use proper/modern parsing techniques someday.

If `set -f` or whatever turns off the globbing (within that function at least?), that might be a good idea, and perhaps Codacy would notice it and stop complaining?

Also the Codacy explanation suggests an alternative: that of using (moving the command-line composition to) a function. It wasn't obvious to me how that might fix the problem, but perhaps it would?

Reply:
Sadly setting `-f` in the shbang didn't convince codacy this is okay... Is it really that important to clear up that warning in codacy? Can't we just ignore it ourselves and merge this anyway, since we know its complaints are meaningless?